comp/serverless/daemon/daemonimpl/daemon_mock.go

```diff
@@ -0,0 +1,100 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build test
+
+package daemonimpl
+
+import (
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/serverless/daemon"
+	"github.com/DataDog/datadog-agent/pkg/serverless/executioncontext"
+	"github.com/DataDog/datadog-agent/pkg/serverless/flush"
+	"github.com/DataDog/datadog-agent/pkg/serverless/invocationlifecycle"
+	serverlessLog "github.com/DataDog/datadog-agent/pkg/serverless/logs"
+	"github.com/DataDog/datadog-agent/pkg/serverless/metrics"
+	"github.com/DataDog/datadog-agent/pkg/serverless/trace"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// MockModule defines the fx options for the mock component.
+var MockModule = fxutil.Component(
+	fx.Provide(newMock),
+)
+
+func newMock(deps dependencies) daemon.Mock {
+	log.Debug("Starting daemon to receive messages from runtime...")
```

Comment:
is all this logic needed in a mock? I thought it could just return the value that are needed for each test

Reply:
I removed all of the unnecessary values from the constructor methods while leaving remaining values that keep the unit tests passing